Logobar theme

malramsay64 opened the pull request:
This implements the logobar theme contributed by jnothman in #5 in a more robust way.

malramsay64 committed:
Create colour definitions for logobar

malramsay64 committed:
Use logo template for logotheme logo

The logo for the logo theme is more representative of placing the logo
in a reasonable position. The titlepage is a special case. With this in
mind the logo template which is used with the \insertlogo command is the
appropriate size for the logobar theme, and the title has a special
template.

malramsay64 committed:
Create outer theme for logobar

This theme defines the footline, frametitle, and headline for the
logobar theme.

malramsay64 committed:
Option for using logobar outer theme

This allows the option logobar to be used in the usetheme directive to
specify the use of the logobar outer theme.

malramsay64 committed:
Expand example to better demonstrate logobar theme

malramsay64 committed:
Make footline transparent on titlepage

The footline is required to be drawn on the titlepage so it has it's
position for the subsequent pages. This is drawn with an empty beamer
colorbox.

malramsay64 committed:
Create placeholder colour for placeholders

I have used the placeholder colour for the items that are placeholders
which is typically on the titlepage. For positioning on subsequent
slides these are required, so I have made them transparent on the
titlepage.

malramsay64 committed:
Include documentation for logobar theme in README

malramsay64 committed:
BUG Use separate colour for non-logobar theme footline

The footline in the non-logobar theme was changed by my changes to the
logobar theme. I have added an additional colour to prevent future
namespace collisions.

malramsay64 committed:
Update example presentations

jnothman reviewed (commented):
I think you need some emphasis in the docs that the dots are produced by use of subsection.

I'll try this out.

jnothman reviewed (commented):
This is nice. :)

I feel like the top margin is slightly excessive, particularly with aspectratio=169, but it's okay

malramsay64 commented on beamerouterthemeusyd-logobar.sty:
```diff
@@ -0,0 +1,57 @@
+
+\defbeamertemplate*{frametitle}{usyd logo}[1][]
+{%
+  \vskip3em
```
I think this value is the top margin you are referring to being too large. Changing this line to `\vskip2em` brings the slide title closer to the logobar.

jnothman replied:
I think this might be too large also, but I was actually referring to the margin north of the logobar

malramsay64 replied:
I have made modifications compressing all the top and bottom components. Is that more appropriate?
<img width="808" alt="screen shot 2018-06-18 at 18 59 37" src="https://user-images.githubusercontent.com/10113564/41527244-d7d453a6-7329-11e8-974b-011dc88130a0.png">

malramsay64 committed:
Reduce spacing between logobar and heading

Particularly on 16:9 slides the real estate being taken up by the
spacing of heading below the logobar was too large. I have changed that
from 3em to 2em.

jnothman commented:
I'm not sure about this... It's now a bit too dense at the top for visual
aesthetics...​ Tricky.

malramsay64 committed:
Compress logobar elements of theme towards edges

This pushes the logo bar elements more towards the outer edge of the
slides, while also slightly shrinking the size of the logobar element.
This is to increase the usable slide real estate, particularly with a
16:9 aspect ratio.

malramsay64 commented:
I'm going to go with splitting the difference between the two. If in using the template you feel quantities need tweaking I am open to minor changes.

malramsay64 committed:
Documentation in README on requiring subsections for dots

The dots in the logobar theme require using subsections, I have offered
two options for achieving this in the README.

malramsay64 committed:
Use \stepcounter over \subsection in example

For the case of the example, using the stepcounter command is more
suitable being a short document. I would also consider it a better
solution than creating arbitratry subsections that aren't actually used.

malramsay64 committed:
Generate both examples with make command

This generates both example presentations using the same make command,
using `sed` to generate the example-logobar.tex file and then compiling
both tex files.

This makes checking changes to the theme significantly easier, since
running `make -W example.tex` regenerates all the necessary files.

malramsay64 committed:
Update example pdf documents

malramsay64 committed:
Include example-logobar.tex in gitignore

This file is generated as part of the make build process.

jnothman commented:
LGTM!

On 19 June 2018 at 15:17, Malcolm Ramsay <notifications@github.com> wrote:

> I'm going to go with splitting the difference between the two. If in using
> the template you feel quantities need tweaking I am open to minor changes.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/malramsay64/usyd-beamer-theme/pull/6#issuecomment-398277070>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz620wGjjihptqTdx46pzZ5YlOMsFNks5t-Il_gaJpZM4UqZkB>
> .
>

jnothman commented:
Great, thanks!​
